src/lib.rs

```diff
@@ -57,6 +57,13 @@ pub mod mem;
 #[cfg(target_arch = "arm")]
 pub mod arm;
 
+#[cfg(all(
+    target_arch = "aarch64",
+    not(feature = "no-asm"),
+    not(feature = "optimized-c")
```

Comment:
note that this means we won't generate the call to the extern symbol in `maybe_use_optimized_c_shim` when `optimized-c` is disabled: https://github.com/rust-lang/compiler-builtins/blob/07cf3b4f109de98910a443c3cacce779eba54cf0/src/macros.rs#L213-L222
i'm not sure how much that matters - it seemed to work ok in core? that whole section of code is very confusing to me, it looks like it's doing an infinitely recursive call but clearly it works *somehow*

Reply:
It's not infinitely recursive  because the outer function is not `no_mangle`. I agree that it's quite confusing though.

Reply:
I think `not(feature = "optimized-c")` should be removed here, the tests rely on these functions being present (it's just that with `optimized-c` they redirect to the compiler-rt version).